pkg/plugins/backendplugin/grpcplugin/client_proto.go

```diff
@@ -66,58 +77,84 @@ func (r *protoClient) Logger() log.Logger {
 }
 
 func (r *protoClient) Start(ctx context.Context) error {
-	return r.plugin.Start(ctx)
+	r.mu.Lock()
+	defer r.mu.Unlock()
+	err := r.plugin.Start(ctx)
+	if err != nil {
+		return err
+	}
+	return nil
 }
 
 func (r *protoClient) Stop(ctx context.Context) error {
-	return r.plugin.Stop(ctx)
+	r.mu.Lock()
+	defer r.mu.Unlock()
+	err := r.plugin.Stop(ctx)
+	if err != nil {
+		return err
+	}
+	return nil
 }
 
-func (r *protoClient) QueryData(ctx context.Context, in *pluginv2.QueryDataRequest, opts ...grpc.CallOption) (*pluginv2.QueryDataResponse, error) {
+func (r *protoClient) client() (*ClientV2, bool) {
```

Comment:
Do you need a read lock/unlock here or is that only needed for start/stop?

Reply:
This is just so a plugin can't be started or stopped at the same time

Reply:
Okay. Can there be a race one starting the plugin and writing `pluginClient` and another one calling this method? In that case read lock/unlock needed was my thinking. But if there's no chance of that you can ignore my comment :)